webxr/headless/mod.rs

```diff
@@ -114,6 +141,26 @@ impl Device for HeadlessDevice {
     fn initial_inputs(&self) -> Vec<InputSource> {
         vec![]
     }
+
+    fn set_event_callback(&mut self, callback: Box<dyn EventCallback>) {
+        self.events.upgrade(callback)
+    }
+
+    fn connected(&mut self) -> bool {
+        if self.connected {
+            true
+        } else {
+            for callback in self.disconnect_callbacks.drain(..) {
+                let _ = callback.send(());
+            }
```

Comment:
Calling this method has a side-effect? I wasn't expecting that!

Reply:
Yeah, this isn't ideal, I want to change this in a followup.